### ISSUE TITLE
Store node: failure to store/load from persistent storage on fleets

### ISSUE THREAD

**Issue by jm-clius** [status-im/nim-waku#791]
## Bug description

The following has been [observed on the `test` fleet](https://grafana.infra.status.im/d/qrp_ZCTGz/nim-waku-v2):
- after a restart the `store` fails to initialize from persistent storage, with log message
```
WRN failed to load messages from store
```
- all nodes report consistently [high store failures](https://grafana.infra.status.im/d/qrp_ZCTGz/nim-waku-v2)

As a result history is lost whenever the fleet gets updated or a node gets restarted.

cc @staheri14 @richard-ramos 

## Related issues

- a PR by @richard-ramos to fix unhelpful error log: https://github.com/status-im/nim-waku/pull/790
- existing issue with `store` response size: https://github.com/status-im/nim-waku/issues/787

**Comment by jm-clius** [comment]
## Why is this happening?

This issue is caused by an inconsistent schema: although the DB `user_version` pragma is set to `2`, the `Message` table schema is still consistent with user_version `1`. In other words, instead of `receiverTimestamp` and `senderTimestamp` columns, the `Message` table still only contains a single ` timestamp` column, causing all read queries (on startup, when the `store` is primed from the persistent DB) _and_ write queries (whenever a new message is stored) to fail. Thanks @richard-ramos for identifying the issue!

## Why is this happening?

It is impossible for the DB `user_version` to be bumped if any of the queries in a migration failed. However, [_it is not currently considered an error or cause to exit migration if a migration finds no query scripts to run in the migration path_](https://github.com/status-im/nim-waku/blob/1217e252458257a8ea2256adaff92fe5df15fa09/waku/v2/node/storage/sqlite.nim#L252). This seems to be what happened on both the `prod` and `test` fleet, with nodes logging:
```
DBG 2021-07-27 07:25:58.078+00:00 scripts to be run                          topics="sqlite" tid=1 file=sqlite.nim:263 scripts=@[]
```
No scripts are run, the migration is considered a success and the `user_version` is bumped to `2`. The next time the node restarts it will determine that no migration is necessary, while queries will continue to fail.

Why these nodes found no `.sql` scripts in the migration path is still unclear. I've tried and failed multiple times to reproduce this original issue locally.

## How should we proceed?

I've [opened a PR](https://github.com/status-im/nim-waku/pull/795) that ensures migration fails if it finds no scripts to run (preventing the `user_version` bump) and adds some debug logging. Once this is merged, I suggest that we:

a) manually revert the `user_version` to `1` for one node in the `test` fleet:
```
sqlite> PRAGMA user_version = 1;
```
b) restart that node and verify that migration was successful and the schema now successful
```
sqlite> .schema Message
```
If it fails again, we'll now have better logs to debug from. If it succeeds, we can follow the same process for the other fleet nodes.

LMK what you think. @staheri14, any theory why the `getScripts()` returned an empty `seq`? This was a long time ago, so may have been a transient issue.

cc @richard-ramos

**Comment by oskarth** [comment]
Nice catch.

> It is impossible for the DB user_version to be bumped if any of the queries in a migration failed. However, it is not currently considered an error or cause to exit migration if a migration finds no query scripts to run in the migration path. This seems to be what happened on both the prod and test fleet, with nodes logging:

Interesting.

> I've opened a PR that ensures migration fails if it finds no scripts to run (preventing the user_version bump) and adds some debug logging. Once this is merged, I suggest that we:

Makes sense.

>  Once this is merged, I suggest that 

Makes sense.

**Comment by jm-clius** [comment]
Issue was caused by lack of migration scripts in docker image. Fix in #797.

**Comment by jm-clius** [comment]
Manually set the `test` fleet `user_version` to `1` and reran migration. Confirmed fixed. Closing this issue.
